backend/Routes/Controllers/group/index.js

```diff
@@ -0,0 +1,167 @@
+const express = require("express");
+const router = express.Router(); 
+const groups = require(".../../../Models/Groups/group-model");
+
+const validator = require("../../../Schema");
+const groupSchema = require("../Schema/groups/groups");
+
+router.post("/addGroup", (req, res) => {
+  let validate = validator.compile(groupSchema.addGroup);
+  let valid = validate(req.body);
+  if (!valid) {
+    res.status(200).json({
+      success: false,
+      error: validate.errors.reduce
+        ? validate.errors.reduce(function(prev, curr) {
+            return curr.message + ";" + prev;
+          }, "")
+        : validate.errors,
+      results: null
+    });
+    return;
+  }
+
+  groups
+    .addGroup(req.body)
+    .then(results => res.json({ success: true, results, error: null }))
+    .catch(error => {
+      res.status(200).json({ success: false, results: null, error });
+    });
+});
+
+router.post("/customGroup", (req, res) => {
+  let validate = validator.compile(groupSchema.customGroup);
+  let valid = validate(req.body);
+  if (!valid) {
+    res.status(200).json({
+      success: false,
+      error: validate.errors.reduce
+        ? validate.errors.reduce(function(prev, curr) {
+            return curr.message + ";" + prev;
+          }, "")
+        : validate.errors,
+      results: null
+    });
+    return;
+  }
+
+  groups
+    .customGroup(req.body)
+    .then(results => res.json({ success: true, results, error: null }))
+    .catch(error => {
+      res.status(200).json({ success: false, results: null, error });
+    });
+});
+
+
+router.get("/getAllGroupsOfUser", function(req, res) {
+  req.query.username = req.body.username;
+  console.log(req.body);
```

Comment:
I guess this was meant for debugging purposes.

Reply:
Fixed